src/datachain/lib/dc.py

```diff
@@ -1976,8 +1981,6 @@ def from_csv(
             else:
                 msg = f"error parsing csv - incompatible output type {type(output)}"
                 raise DatasetPrepareError(chain.name, msg)
-        elif nrows:
-            nrows += 1
```

Comment:
This was a workaround to make `nrows` works if the data contains headers. This is no longer needed as we count the rows internally.